src/jit/morph.cpp

```diff
@@ -7309,7 +7403,10 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
         }
 #endif // FEATURE_TAILCALL_OPT
 
-        fgFixupStructReturn(call);
+        if (varTypeIsStruct(call))
+        {
+            fgFixupStructReturn(call);
+        }
 
```

Comment:
Maybe the first line of fgFixupStructReturn should be "if (!varTypeIsStruct(call)) return;"?